pkg/scalers/datadog_scaler.go

```diff
@@ -0,0 +1,316 @@
+			meta.datadogSite = val
+		} else {
+			meta.datadogSite = "datadoghq.com"
+		}
+	} else {
+		meta.datadogSite = "datadoghq.com"
+	}
+
+	return &meta, nil
+}
+
+// newDatddogConnection tests a connection to the Datadog API
+func newDatadogConnection(ctx context.Context, meta *datadogMetadata) (*datadog.APIClient, error) {
+
+	ctx = context.WithValue(
+		context.Background(),
+		datadog.ContextAPIKeys,
+		map[string]datadog.APIKey{
+			"apiKeyAuth": {
+				Key: meta.apiKey,
+			},
+			"appKeyAuth": {
+				Key: meta.appKey,
+			},
+		},
+	)
+
+	ctx = context.WithValue(ctx,
+		datadog.ContextServerVariables,
+		map[string]string{
+			"site": meta.datadogSite,
+		})
+
+	configuration := datadog.NewConfiguration()
+	apiClient := datadog.NewAPIClient(configuration)
+
+	_, _, err := apiClient.AuthenticationApi.Validate(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to Datadog API endpoint: %v", err)
+	}
+
+	return apiClient, nil
+}
+
+// No need to close connections
+func (s *datadogScaler) Close(context.Context) error {
+	return nil
+}
+
+// IsActive returns true if we are able to get metrics from Datadog
+func (s *datadogScaler) IsActive(ctx context.Context) (bool, error) {
+
+	ctx = context.WithValue(
+		context.Background(),
+		datadog.ContextAPIKeys,
+		map[string]datadog.APIKey{
+			"apiKeyAuth": {
+				Key: s.metadata.apiKey,
+			},
+			"appKeyAuth": {
+				Key: s.metadata.appKey,
+			},
+		},
+	)
+
+	ctx = context.WithValue(ctx,
+		datadog.ContextServerVariables,
+		map[string]string{
+			"site": s.metadata.datadogSite,
+		})
+
+	resp, _, err := s.apiClient.MetricsApi.QueryMetrics(ctx, time.Now().Unix()-int64(s.metadata.age), time.Now().Unix(), s.metadata.query)
+
+	if err != nil {
+		return false, err
+	}
+
+	series := resp.GetSeries()
+
+	if len(series) == 0 {
+		return false, nil
+	}
+
+	points := series[0].GetPointlist()
+
+	if len(points) == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// getQueryResult returns result of the scaler query
+func (s *datadogScaler) getQueryResult(ctx context.Context) (int, error) {
+
+	ctx = context.WithValue(
+		context.Background(),
+		datadog.ContextAPIKeys,
+		map[string]datadog.APIKey{
+			"apiKeyAuth": {
+				Key: s.metadata.apiKey,
+			},
+			"appKeyAuth": {
+				Key: s.metadata.appKey,
+			},
+		},
+	)
+
+	ctx = context.WithValue(ctx,
+		datadog.ContextServerVariables,
+		map[string]string{
+			"site": s.metadata.datadogSite,
+		})
+
+	resp, _, err := s.apiClient.MetricsApi.QueryMetrics(ctx, time.Now().Unix()-int64(s.metadata.age), time.Now().Unix(), s.metadata.query)
+	if err != nil {
+		return -1, fmt.Errorf("error when retrieving Datadog metrics: %s", err)
+	}
+
+	series := resp.GetSeries()
+
+	if len(series) == 0 {
+		return 0, fmt.Errorf("no Datadog metrics returned")
+	}
+
+	points := series[0].GetPointlist()
+
+	if len(points) == 0 {
+		return 0, fmt.Errorf("no Datadog metrics returned")
+	}
+
+	var val int = int(*points[0][1])
+
+	return val, nil
```

Comment:
do we need to check whether `*points[0]` has 2 (exactly, or more than) elements in it? WDYT of something like the below?

```suggestion
	if len(points) == 0 || len(points[0]) <= 2 {
		return 0, fmt.Errorf("no Datadog metrics returned")
	}

	return int(*points[0][1]), nil
```